### ISSUE TITLE
Total number of annotations cannot be higher than the file duration in seconds

### ISSUE THREAD

**Issue by josefransaenz** [holgern/pyedflib#34]
Hi, 
It seems that the maximum number of annotations that can be written in a EDF+ file with this library is equal to the duration in seconds of the signals. Why?
I have done several tests and for files with 10s, 100s, 1000s or 10000s long records it's not possible to write more than 10, 100, 1000 or 10000 annotations respectively. There is no error raised by the f.writeAnnotation method, I just found it out after opening the newly created file where only the first N=records_duration annotations where present. I don't know if this is a limit of the EDF format but if so, it should be documented and the f.writeAnnotation method should return an error code if a limit is reached.
Best,
Jose

**Comment by cboulay** [comment]
I've encountered this too. It's not strictly 1 per second. The limit is 1 annotation per "record", and the default record size is 1 second. You can change the record size, and thus increase the number of annotations per second, but I've found this to be very buggy, and increasing the total number of records really slows down file loading.

Alternatively, if you're using EDF+ or BDF+, you can [`set_number_of_annotation_signals`](https://github.com/holgern/pyedflib/blob/master/pyedflib/edfwriter.py#L364-L381) to 64. This will allow for up to 64 annotations per record.

Note that a data record may not exceed 61440 bytes, including the annotation signal(s).
64 annot /s * 114 bytes / annot = 7296 bytes, leaving 54144 bytes of _data_ per record.

**Comment by cboulay** [comment]
I agree that it would be nice if there was a warning to tell you that your annotations will be truncated. Raising an error is a bad idea because the problem won't be known until file close. e.g., If your first 1 second of data has 100 annotations, but your next 99 seconds of data have 0 annotations, then you're still OK (I think?). But if you only have 98 more seconds, then your last annotation will be lost.

Maybe an error can be raised after file close.
